modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/load/IgniteCacheRandomOperationBenchmark.java

```diff
@@ -713,229 +846,283 @@ private void doSqlQuery(IgniteCache cache) throws Exception {
     }
 
     /**
-     * Closure for scan query executing.
+     * @param cache Ignite cache.
+     * @param map Parameters map.
+     * @throws Exception If failed.
      */
-    private static class ScanQueryBroadcastClosure implements IgniteRunnable {
-        /**
-         * Ignite node.
-         */
-        @IgniteInstanceResource
-        private Ignite node;
+    private void doContinuousQuery(IgniteCache cache, Map<Object, Object> map) throws Exception {
+        QueryCursor cursor = (QueryCursor)map.get(cache.getName());
+        if (cursor != null)
+            cursor.close();
```

Comment:
There is a high chance that there query will be started and stopped too frequently. I would allow to have up to 3 cont queries running for the same cache. When 4-th query is started you can randomly close any previous one,